### ISSUE TITLE
Funimation: Not finding all available subtitle languages for certain episodes 

### ISSUE THREAD

**Issue by ShadowReviews** [yt-dlp/yt-dlp#1656]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.11.10.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Description

I'm trying to download episodes of Vivy from Funimation, and for episodes 1 and 2, there are no English subtitles available. On the Funimation website, there are English subtitles, but when using `--list-subs`, only Portuguese subtitles are displayed and using `--sub-langs en,enUS` results in no subtitles.

Weirdly enough, episodes 3-14 listed English subtitles and downloaded them, though they didn't embed due to the Funimation embedded subtitles issue that's currently open and known. I don't see any differences in the video versions on the site between episodes 1 and 3 or when using the `-F` flag to list all formats that would indicate there should be any change in behavior. Both have only simulcast versions available and are available in English and Japanese with three subtitle languages available.

Verbose logs for downloading both episode 1 and episode 3 are included below, but using the `--list-subs` flag. Episode 1 and 2 only list Portuguese subs, but all other episodes list English, Spanish, and Portuguese (or at least English). I tried to identify the issue but haven't been able to figure anything out so far.

When you look at the show page for Vivy, where it lists the languages under the episode title, episodes 1 and 2 say 'English, Japanese,' while the rest of the episodes say 'Japanese, English'. Horimiya displays this same behavior but English subs are still available for episodes 1 and 2, so that doesn't seem to be the issue. I also tried adding `/?qid=` and `lang=` flags to the URL but that didn't seem to change anything either. I'll try looking at other shows on the site later today to see if there's more to this issue or it's somehow specific to two episodes of a single show.

EDIT: I can go back later and grab the output of `-F` and include that in the log if it'll help.

### Verbose log

```shell
Episode 1: 

[debug] Command-line config: ['-f', 'bv*[height=1080]+ba', 'https://www.funimation.com/v/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing', '--username', 'PRIVATE', '--password', 'PRIVATE', '--cookies', 'funimation_cookies.txt', '--extractor-args', 'funimation:language=japanese,version=simulcast', '-N', '64', '-o', 'S%(season_number)s_E%(episode_number)s_%(episode)s.mp4', '--verbose', '--embed-subs', '--sub-langs', 'en,enUS', '--list-subs']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.11.10.1
[debug] Lazy loading extractors is disabled
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.14.16-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[funimation:page] Logging in
[debug] [funimation:page] Extracting URL: https://www.funimation.com/v/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing
[funimation:page] vivy-fluorite-eyes-song-_my-code-to-make-everyone-happy-with-my-singing: Downloading JSON metadata
[debug] [Funimation] Extracting URL: https://www.funimation.com/player/1523340
[Funimation] 1523340: Downloading player webpage for 1523340
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading player webpage for Simulcast japanese (1397018)
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) JSON
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) m3u8 information
[debug] Sort order given by extractor: lang, source
[debug] Formats sorted by: hasvid, ie_pref, lang, source, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, id
Available subtitles for 1394903:
Language Name                          Formats
pt       Simulcast Portuguese (Brazil) srt, vtt


Episode 3:

[debug] Command-line config: ['-f', 'bv*[height=1080]+ba', 'https://www.funimation.com/v/vivy-fluorite-eyes-song-/a-tender-moon-tempo-a-pleasant-chat-with-the-stars', '--username', 'PRIVATE', '--password', 'PRIVATE', '--cookies', 'funimation_cookies.txt', '--extractor-args', 'funimation:language=japanese,version=simulcast', '-N', '64', '-o', 'S%(season_number)s_E%(episode_number)s_%(episode)s.mp4', '--verbose', '--embed-subs', '--sub-langs', 'en,enUS', '--list-subs']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.11.10.1
[debug] Lazy loading extractors is disabled
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.14.16-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[funimation:page] Logging in
[debug] [funimation:page] Extracting URL: https://www.funimation.com/v/vivy-fluorite-eyes-song-/a-tender-moon-tempo-a-pleasant-chat-with-the-stars
[funimation:page] vivy-fluorite-eyes-song-_a-tender-moon-tempo-a-pleasant-chat-with-the-stars: Downloading JSON metadata
[debug] [Funimation] Extracting URL: https://www.funimation.com/player/1594334
[Funimation] 1594334: Downloading player webpage for 1594334
[Funimation] a-tender-moon-tempo-a-pleasant-chat-with-the-stars: Downloading player webpage for Simulcast japanese (1403013)
[Funimation] a-tender-moon-tempo-a-pleasant-chat-with-the-stars: Downloading Simulcast japanese (1403013) JSON
[Funimation] a-tender-moon-tempo-a-pleasant-chat-with-the-stars: Downloading Simulcast japanese (1403013) m3u8 information
[debug] Sort order given by extractor: lang, source
[debug] Formats sorted by: hasvid, ie_pref, lang, source, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, id
[debug] Downloading subtitles: en
Available subtitles for 1396401:
Language Name                          Formats
en       Simulcast English             srt, vtt
es       Simulcast Spanish (Latin Am)  srt, vtt
pt       Simulcast Portuguese (Brazil) srt, vtt
```

**Comment by pukkandan** [comment]
Related https://www.reddit.com/r/youtubedl/comments/qsuxke/funimation_english_subs_not_available_for_episode/

**Comment by pukkandan** [comment]
For devs:

These subs are not available in the /player/ URLs and therefore does not appear in the API calls yt-dlp makes. It also does not appear in the API endpoint @Mevious suggested in https://github.com/yt-dlp/yt-dlp/issues/428#issuecomment-868134291.

The existence of these subs are mentioned in `https://d33et77evd9bgg.cloudfront.net/data/v1/episodes/my-code-to-make-everyone-happy-with-my-singing.json` endpoint and the URLs can be obtained from `https://playback.prd.funimationsvc.com/v1/play/VIV0001?deviceType=web&playbackStreamId=5e73c34d-205f-4511-9c5b-efeb85e21e71`. However, the `playbackStreamId` needed for this endpoint seems to be tied to the session and we will need to reverse engineer how it is generated

**Comment by Jules-A** [comment]
Only way I can think of is to get the URI from the video manifest for example: https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6out.key removing out.key and appending _english_full.vtt or _spanishlatinamerica_full.vtt ect. for each subtitle type listed from the api e.g. https://prod-api-funimationnow.dadcdigital.com/api/source/catalog/episode/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing/ under items>0>siblings>{episode num}>languages or using another api so you end up with https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_english_full.vtt , https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_spanishlatinamerica_full.vtt  ect.

**Comment by ScottSemian** [comment]
> Only way I can think of is to get the URI from the video manifest for example: `https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6out.key` removing out.key and appending _english_full.vtt or _spanishlatinamerica_full.vtt ect. for each subtitle type listed from the api e.g. https://prod-api-funimationnow.dadcdigital.com/api/source/catalog/episode/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing/ under items>0>siblings>{episode num}>languages or using another api so you end up with https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_english_full.vtt , https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_spanishlatinamerica_full.vtt ect.

I was able to use this method to basically brute force and find the second subtitle file for episode two.

The API page for episode 2 is https://prod-api-funimationnow.dadcdigital.com/api/source/catalog/episode/vivy-fluorite-eyes-song-/quarter-note-the-beginning-of-the-one-hundred-year-journey/

And the .vtt file is located at https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272751/06b23e58-e37e-46dc-8f75-08044f13b4b1_english_full.vtt

If that was the solution it appears it would be replicable in any instance where it occurred provided you can get the correct key reliably, I'm not familiar with the extractor so maybe there's some way of telling them apart I just wasn't able to figure out.

**Comment by Shadow5435** [comment]
> Only way I can think of is to get the URI from the video manifest for example: `https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6out.key` removing out.key and appending _english_full.vtt or _spanishlatinamerica_full.vtt ect. for each subtitle type listed from the api e.g. https://prod-api-funimationnow.dadcdigital.com/api/source/catalog/episode/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing/ under items>0>siblings>{episode num}>languages or using another api so you end up with https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_english_full.vtt , https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6_spanishlatinamerica_full.vtt ect.

I was able to use this method to basically brute force and find the second subtitle file for episode two.

The API page for episode 2 is https://prod-api-funimationnow.dadcdigital.com/api/source/catalog/episode/vivy-fluorite-eyes-song-/quarter-note-the-beginning-of-the-one-hundred-year-journey/

And the .vtt file is located at https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272751/06b23e58-e37e-46dc-8f75-08044f13b4b1_english_full.vtt

If that was the solution it appears it would be replicable in any instance where it occurred provided you can get the correct key reliably, I'm not familiar with the extractor so maybe there's some way of telling them apart I just wasn't able to figure out.

**Comment by Jules-A** [comment]
> If that was the solution it appears it would be replicable in any instance where it occurred provided you can get the correct key reliably, I'm not familiar with the extractor so maybe there's some way of telling them apart I just wasn't able to figure out.

I think there is other ways to get that part of the url (```https://d33et77evd9bgg.cloudfront.net/FunimationStoreFront/V2272750/f586e90e-7684-4a71-8b80-e874732bbbb6```), that was just the first way I thought of getting it, I'm sure there's other ways.

**Comment by CZeil** [comment]
@ShadowReviews it would be nice if you could go back to see if the subtitles of the video appear. Here is the verbose log that I just extracted from the first chapter of vivy.

```
C:\youtube-dl>yt-dlp --list-subs --cookies cookies-funimation-com.txt https://www.funimation.com/v/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing
[funimation:page] vivy-fluorite-eyes-song-_my-code-to-make-everyone-happy-with-my-singing: Downloading JSON metadata
[Funimation] 1523340: Downloading player webpage for 1523340
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading player webpage for Simulcast japanese (1397018)
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) JSON
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) m3u8 information
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast english (1523340) JSON
[Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast english (1523340) m3u8 information
Available subtitles for 1394903:
Language Name                          Formats
en       Simulcast English             srt, vtt
es       Simulcast Spanish (Latin Am)  srt, vtt
pt       Simulcast Portuguese (Brazil) srt, vtt
```

**Comment by ShadowReviews** [comment]
> @ShadowReviews it would be nice if you could go back to see if the subtitles of the video appear. Here is the verbose log that I just extracted from the first chapter of vivy.
> 
> ```
> C:\youtube-dl>yt-dlp --list-subs --cookies cookies-funimation-com.txt https://www.funimation.com/v/vivy-fluorite-eyes-song-/my-code-to-make-everyone-happy-with-my-singing
> [funimation:page] vivy-fluorite-eyes-song-_my-code-to-make-everyone-happy-with-my-singing: Downloading JSON metadata
> [Funimation] 1523340: Downloading player webpage for 1523340
> [Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading player webpage for Simulcast japanese (1397018)
> [Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) JSON
> [Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast japanese (1397018) m3u8 information
> [Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast english (1523340) JSON
> [Funimation] my-code-to-make-everyone-happy-with-my-singing: Downloading Simulcast english (1523340) m3u8 information
> Available subtitles for 1394903:
> Language Name                          Formats
> en       Simulcast English             srt, vtt
> es       Simulcast Spanish (Latin Am)  srt, vtt
> pt       Simulcast Portuguese (Brazil) srt, vtt
> ```

I just went back and checked and it looks like they are there now. I've started having other issues with the Funimation extractor unrelated to this issue though. I suppose for now we could mark this issue as closed and revisit it in the future if needed, maybe the answer to get missing subtitles to appear is to check back in a few days.
